Assets/MixedRealityToolkit-SDK/Features/Input/Handlers/ManipulationHandler.cs

```diff
@@ -0,0 +1,489 @@
+            Rotating = 0x100,
+            MovingScaling = 0x011,
+            RotatingScaling = 0x110,
+            MovingRotatingScaling = 0x111
+        };
+        #endregion Fields
+
+        #region Private Properties
+        /// <summary>
+        /// private properties that store transform information.
+        /// </summary>
+        private State currentState;
+        private TwoHandMoveLogic m_moveLogic;
+        private TwoHandScaleLogic m_scaleLogic;
+        private TwoHandRotateLogic m_rotateLogic;
+        private GameObject cubeLeft;
+        private GameObject cubeRight;
+        private bool usingPose = false;
+
+        [SerializeField]
+        private TextMesh debugText;
+
+        /// <summary>
+        /// Maps input id -> position of hand
+        /// </summary>
+        private readonly Dictionary<uint, Vector3> m_handsPressedLocationsMap = new Dictionary<uint, Vector3>();
+
+        /// <summary>
+        /// Maps input id -> input source. Then obtain position of input source using currentInputSource.TryGetGripPosition(currentInputSourceId, out inputPosition);
+        /// </summary>
+        private readonly Dictionary<uint, IMixedRealityInputSource> m_handsPressedInputSourceMap = new Dictionary<uint, IMixedRealityInputSource>();
+
+        private readonly Dictionary<uint, Vector3> m_handsPoseMap = new Dictionary<uint, Vector3>();
+        #endregion
+
+        #region Monobehaviour Functions
+        /// <summary>
+        /// Private Methods
+        /// </summary>
+        private void Awake()
+        {
+            m_moveLogic = new TwoHandMoveLogic();
+            m_rotateLogic = new TwoHandRotateLogic(constraintOnRotation);
+            m_scaleLogic = new TwoHandScaleLogic();
+        }
+
+        private void Start()
+        {
+            if (hostTransform == null)
+            {
+                hostTransform = transform;
+            }
+        }
+
+        private void Update()
+        {
+            //Update positions of all hands
+            foreach (var key in m_handsPressedInputSourceMap.Keys)
+            {
+                var inputSource = m_handsPressedInputSourceMap[key];
+                Vector3 inputPosition = Vector3.zero;
+
+                if (inputSource.SourceName.Contains("Hand") == false)
+                {
+                    inputSource.Pointers[0].TryGetPointerPosition(out inputPosition);
+                }
+                else
+                {
+                   inputPosition = m_handsPoseMap[key];
+                }
+
+                m_handsPressedLocationsMap[key] = inputPosition;
+            }
+
+            if (currentState != State.Start)
+            {
+                UpdateStateMachine();
+            }
+        }
+        #endregion Monobehaviour Functions
+
+        #region Private Methods
+        /// <summary>
+        /// SetManipulationMode
+        /// </summary>
+        private void SetManipulationMode(TwoHandedManipulation mode)
```

Comment:
should this be a private set property instead of a method?

Reply:
removed- wasn't being used